### DISCUSSION TITLE
headerSearchBarOptions, set the search input value

### DISCUSSION THREAD

**Discussion by hamdij0maa** [software-mansion/react-native-screens#1737] [Ideas]
headerSearchBarOptions, set the search input value, 
there is no function exposed to set the value of the search bar from outside the onTextChangeValue, 
example => sending the value from the search screen 1 to the second screen 2, is impossible now. 
any solution please

**Comment by kkafar** [comment]
Hi @hamdij0maa, could you elaborate a little bit more of what would be the expected functionality (what API you would like to see)? You receive current searchbar text & you can receive event when search button is clicked -> you can save the search bar content to state.

**Reply by hamdij0maa** [reply]
Hello @kkafar, It will be helpfull if we can set the value of the search bar from outside those events, 
setSearchBarValue('new value'), 
example => instagram search interface on search button pressed it navigates to an other screen with the same value already typed. 
if we do it using rn screens now, the value will be empty cause we can not set it without using the keyboard input events

**Reply by kkafar** [reply]
Oh, I get it know -- that idea emerged also [here](https://github.com/software-mansion/react-native-screens/discussions/1617) and [here](https://github.com/software-mansion/react-native-screens/discussions/1323#discussioncomment-4282479), what's more -- it is somewhere in my backlog --> it will surely be done, but I won't declare any timelines. Any PR's are welcome if you're feeling like getting your hands dirty :)

**Reply by hamdij0maa** [reply]
@kkafar  it will be a pleasure, I will try to make it happen as soon as I can 
thanks for being always here man 🚀

**Reply by kkafar** [reply]
@hamdij0maa, unless you are already on this -- I've changed my mind as it won't take much time to implement -- I'll do it right now. You can track the progress and leave comments [here](https://github.com/software-mansion/react-native-screens/pull/1739)

**Reply by hamdij0maa** [reply]
great brother thank you ok am in I'll keep u informed too, 🚀

**Reply by kkafar** [reply]
Would love to get some feedback: https://github.com/software-mansion/react-native-screens/pull/1739 whether the `setText` method is enough or we need some additional mechanisms

You can install react-native-screens via:

```json
"react-native-screens": "software-mansion/react-native-screens#@kkafar/search-bar-set-text"
```

& play around.

You can find example API usage [here](https://github.com/software-mansion/react-native-screens/blob/2a30e22bf1bd432fdd5e2f4d44e42bc3efd44474/TestsExample/src/Test1097.tsx#L137-L201)

**Reply by hamdij0maa** [reply]
okay great thanks man,  🎉

**Reply by hamdij0maa** [reply]
@kkafar great job :D when it will be released ? any time soon ?

**Reply by kkafar** [reply]
As I said [here](https://github.com/software-mansion/react-native-screens/pull/1728#issuecomment-1469916746) -- the timeline is blurry and dynamic -- I don't want to set any specific expectations.
